vscode-ext/src/StatusBarComponent.ts

```diff
@@ -38,36 +47,138 @@ export class StatusBarComponent {
 
         // listen to verifier updates
         verifier.client.onStatusUpdate((status) => {
-            this.status = status;
+            this.serverStatus = status;
+            this.render();
+        });
+
+        verifier.client.onVerifyResult((document, results) => {
+            this.verifyStatus.insert(document, results);
             this.render();
         });
+
     }
 
     render() {
         if (this.enabled) {
-            switch (this.status) {
+
+            switch (this.serverStatus) {
+                case ServerStatus.NotStarted:
+                    this.view.text = "$(debug-start) Caesar Inactive";
+                    this.view.command = "caesar.startServer";
+                    this.tooltipBase = stoppedTooltipBase;
+                    break;
                 case ServerStatus.Stopped:
-                    this.view.text = "$(debug-stop) Et tu, Brute?";
+                    this.view.text = "$(debug-start) Et tu, Brute?";
+                    this.view.command = "caesar.startServer";
+                    this.tooltipBase = stoppedTooltipBase;
                     break;
                 case ServerStatus.FailedToStart:
                     this.view.text = "$(warning) Failed to start Caesar";
+                    this.view.command = "caesar.startServer";
+                    this.tooltipBase = stoppedTooltipBase;
                     break;
                 case ServerStatus.Starting:
-                    this.view.text = "$(sync~spin) Starting Caesar...";
+                    this.view.text = "$(loading~spin) Starting Caesar...";
+                    this.view.command = ""
                     break;
                 case ServerStatus.Ready:
-                    this.view.text = "$(check) Caesar Ready";
+                    this.view.text = "$(thumbsup-filled) Caesar Ready";
+                    this.view.command = "caesar.showOutput";
+                    this.tooltipBase = runningTooltipBase;
                     break;
                 case ServerStatus.Verifying:
                     this.view.text = "$(sync~spin) Verifying...";
+                    this.view.command = ""
                     break;
                 case ServerStatus.Finished:
-                    this.view.text = "$(check) Verified";
+                    this.tooltipBase = runningTooltipBase;
+                    this.handleFinishedStatus();
+                    this.view.command = "caesar.showOutput";
                     break;
             }
+
+            this.renderTooltip();
             this.view.show();
         } else {
             this.view.hide();
         }
     }
+
+    private extendedTooltip(extension: vscode.MarkdownString) {
```

Comment:
Seems like there is a return type missing? I wonder why TypeScript doesn't complain about this.

Also, I think this method should be inlined. As far as I can tell, it's just used once.